File: 1. JPA basic/basic/src/main/java/study/jpa/basic/jpashop/domain/Movie.java
package study.jpa.basic.jpashop.domain;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * @author Joonhyuck Hyoung
 */
@Entity(name = "jpashop_movie")
@Table(name = "jpashop_movie")
public class Movie extends Item {
    private String director;
    private String actor;

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public String getActor() {
        return actor;
    }

    public void setActor(String actor) {
        this.actor = actor;
    }
}

File: 2. JPA deep/jpashop/src/main/kotlin/jpabook/jpashop/domain/item/Movie.kt
package jpabook.jpashop.domain.item

import javax.persistence.*

/**
 * @author Joonhyuck Hyoung
 */
@Entity
@DiscriminatorValue("MOVIE")
class Movie(
    val director: String,
    val actor: String,

    name: String,
    price: Int,
    stockQuantity: Int,
) : Item(name, price, stockQuantity)
